proto/src/determined/api/v1/trial.proto

```diff
@@ -44,6 +45,324 @@ message WorkloadContainer {
   }
 }
 
+// A filter structure to allow for range filters on arbitrary keys
+message DoubleRangeFilter {
```

Comment:
there's a parallel discussion about filters here: https://github.com/determined-ai/determined/pull/4925#discussion_r966373082

I think this is hitting the same question: we need a unified way for standard filters of fields. This min/max solution is basically `__gte` / `__lte`, in addition to `_in` / `_notin` in Keita's PR.